### PR TITLE
feat(rpc): add a method to remove a transaction from tx-pool

### PR DESCRIPTION
**Pull request by yangby-cryptape**
### What problem does this PR solve?

If there are some transactions in tx pool, but we want to delete them, at present, the only way is to clear the whole tx pool.

This PR add a new RPC to remove a transaction from tx pool via its hash.

### Check List

Tests

- Integration test

### Release note

```release-note
Title Only: Include only the PR title in the release note.
```

### CODE REVIEW

**Review by chanhsu001 — COMMENTED** [review]
the RPC mainly serves "clear some transaction(big cycle transaction, etc.)" and "single node test"

**Comment by yangby-cryptape** [comment]
> the RPC mainly serves "clear some transaction(big cycle transaction, etc.)" and "single node test"

This RPC only provides a new operation, how to use this RPC based on people's imagination.

There is no limit.

**Review by quake — APPROVED** [review]

File: tx-pool/src/pool.rs [hunk 1]
```diff
@@ -342,6 +342,30 @@ impl TxPool {
...
+    pub(crate) fn remove_tx(&mut self, id: &ProposalShortId) -> bool {
+        let mut entries = self.proposed.remove_entry_and_descendants(id);
+        if !entries.is_empty() {
+            entries.sort_unstable_by_key(|entry| entry.ancestors_count);
+            for mut entry in entries {
+                entry.reset_ancestors_state();
+                self.update_statics_for_remove_tx(entry.size, entry.cycles);
+            }
+            return true;
+        }
```
Comment (zhangsoledad) [inline tx-pool/src/pool.rs:354]:
``` suggestion
        let entries = self.proposed.remove_entry_and_descendants(id);
        if !entries.is_empty() {
            for entry in entries {
                self.update_statics_for_remove_tx(entry.size, entry.cycles);
            }
            return true;
        }
```

Comment (zhangsoledad) [reply tx-pool/src/pool.rs:354]: sort and reset is useless for removing

**Review by zhangsoledad — APPROVED** [review]

**Comment by zhangsoledad** [comment]
bors r= quake,zhangsoledad

**Comment by yangby-cryptape** [comment]
bors r= quake,zhangsoledad

**Pull request merged**